Data/ODBC/include/Poco/Data/ODBC/Extractor.h

```diff
@@ -53,7 +55,8 @@ class ODBC_API Extractor: public Poco::Data::AbstractExtractor
 	typedef Preparator::Ptr PreparatorPtr;
 
 	Extractor(const StatementHandle& rStmt, 
-		Preparator::Ptr pPreparator);
+		Preparator::Ptr pPreparator,
+		const std::string& dbEncoding = "UTF-8");
```

Comment:
I would pass in the Encoding object (Ptr) directly, to avoid having to look it up every time. Encoding objects are immutable, so this is safe.

Reply:
I've done something along those lines, but on the constructor side, see https://github.com/pocoproject/poco/pull/3146/commits/cf2d7c75c10a639b3ebf17e44285ec2ee3ed8c32

Reply:
probably also a good idea to verify that the encoding is valid early (ie. immediately when it is set), I'll add that check